src/scheduler/annualReviewUnpublishedEmailScheduler.ts

```diff
@@ -0,0 +1,293 @@
+   unpublish date ${unpublishedDateContext.unpublishDate.toUTCString()}
+   annual review start date ${unpublishedDateContext.annualReviewStartDate}`);
+
+  const listIds = lists.map(list => list.id);
+  const findListItemsResult = await findListItemsForLists(listIds, [Status.CHECK_ANNUAL_REVIEW], true);
+
+  if (findListItemsResult.error) {
+    logger.error(`Could not send unpublished emails - ListItems could not be retrieved`);
+
+  } else if (findListItemsResult.result) {
+    const listItemsForAllLists = findListItemsResult.result;
+    for (const list of lists) {
+      const listItemsEligibleForAnnualReview = listItemsForAllLists.filter((listItem) => listItem.listId === list.id);
+
+      // only send emails if there are list items still in CHECK_ANNUAL_REVIEW status
+      if (listIds.length > 0) {
+        await sendPostEmails(unpublishedDateContext, list, listItemsEligibleForAnnualReview);
+
+        // update list items and email providers to confirm annual review start
+        for (const listItem of listItemsEligibleForAnnualReview) {
+          let updatedListItem;
+
+          if (daysBeforeUnpublishing === TODAY) {
+            updatedListItem = await updateUnpublished(
+              listItem,
+              Status.UNPUBLISHED,
+              ListItemEvent.UNPUBLISHED,
+              AuditEvent.UNPUBLISHED
+            );
+
+            if (!updatedListItem) {
+              logger.info(`List items ${listItem.id} could not be updated`);
+            }
+          }
+          if (daysBeforeUnpublishing !== 0 || (daysBeforeUnpublishing === 0 && updatedListItem)) {
+            await sendProviderEmails(unpublishedDateContext, list, listItem);
+          }
+        }
+      }
+    }
+  }
+}
+
+interface UnpublishedDateContext {
+  annualReviewStartDate: Date;
+  daysBeforeUnpublished: number;
+  unpublishDate: Date;
+}
+
+/**
+ * Calculate the annual review dates six weeks prior to the unpublish date of 0, 1, 7, 14, 21, 28, 35 days in the future
+ * from the todayDateString.
+ * @param todayDateString
+ * @returns UnpublishedDateContext[]
+ */
+function getUnpublishedDateContexts(todayDateString: string): UnpublishedDateContext[] {
+  const today = getTodayDate(todayDateString);
+  const unpublishedDateSixWeeksAway = addDays(today, SIX_WEEKS);
+  const unpublishedDateOneDayAway = subDays(unpublishedDateSixWeeksAway, 41);
+
+  const unpublishedDateContextsForFiltering: UnpublishedDateContext[] = [
+    {
+      annualReviewStartDate: subDays(unpublishedDateOneDayAway, SIX_WEEKS),
+      daysBeforeUnpublished: ONE_DAY,
+      unpublishDate: unpublishedDateOneDayAway,
+    },
+    {
+      annualReviewStartDate: subDays(today, SIX_WEEKS),
+      daysBeforeUnpublished: 0,
+      unpublishDate: today,
+    },
+  ];
+
+  // fill in the dates between 1 to 5 weeks from the todayDateString
+  for (let daysBeforeUnpublished = ONE_WEEK; daysBeforeUnpublished <= FIVE_WEEKS; daysBeforeUnpublished += 7) {
```

Comment:
wow, such an old school loop lol (imo the most unreadable.. I always have to look up what in the fresh hell is going on here)